### ISSUE TITLE
Can't change crs

### ISSUE THREAD

**Issue by aijanai** [geopandas/geopandas#990]
Hi,
I can't make this toy example  work:

``` 
from shapely.geometry import Point

s = geopandas.GeoSeries([Point(4.42, 50.4), Point(4.43, 50.2)])
s.crs = {'init': 'epsg:4326', 'no_defs': True}
s.to_crs(epsg=31370)
```

It yelds to:

```
---------------------------------------------------------------------------
CRSError                                  Traceback (most recent call last)
<ipython-input-153-c63352d64344> in <module>
      1 s = geopandas.GeoSeries([Point(4.42, 50.4), Point(4.43, 50.2)])
      2 s.crs = {'init': 'epsg:4326', 'no_defs': True}
----> 3 s.to_crs(epsg=31370)

/opt/conda/lib/python3.7/site-packages/geopandas/geoseries.py in to_crs(self, crs, epsg)
    302             except TypeError:
    303                 raise TypeError('Must set either crs or epsg for output.')
--> 304         proj_in = pyproj.Proj(self.crs, preserve_units=True)
    305         proj_out = pyproj.Proj(crs, preserve_units=True)
    306         if _PYPROJ2:

/opt/conda/lib/python3.7/site-packages/pyproj/proj.py in __init__(self, projparams, preserve_units, **kwargs)
    293         '116.366 39.867'
    294         """
--> 295         self.crs = CRS.from_user_input(projparams if projparams is not None else kwargs)
    296         # make sure units are meters if preserve_units is False.
    297         if not preserve_units and "foot" in self.crs.axis_info[0].unit_name:

/opt/conda/lib/python3.7/site-packages/pyproj/crs.py in from_user_input(cls, value)
    222             return cls.from_epsg(value)
    223         elif isinstance(value, dict):
--> 224             return cls(**value)
    225         elif isinstance(value, string_types):
    226             return cls.from_string(value)

/opt/conda/lib/python3.7/site-packages/pyproj/crs.py in __init__(self, projparams, **kwargs)
    144         # on case-insensitive filesystems).
    145         projstring = projstring.replace("+init=EPSG", "+init=epsg").strip()
--> 146         super(CRS, self).__init__(projstring)
    147 
    148     @classmethod

pyproj/_crs.pyx in pyproj._crs._CRS.__init__()

CRSError: Invalid projection: +init=epsg:4326 +no_defs +type=crs: (Internal Proj Error: proj_create: cannot expand +init=epsg:4326 +no_defs +type=crs)
```

What am I doing wrong? 

Using:

- Python 3.7.3
- Jupyter Docker image jupyter/scipy-notebook latest 
-  conda 4.6.14
-  pyproj '2.1.3'

**Comment by aijanai** [comment]
SOLVED

It was pyproj 2.1.3. By setting it to 1.9.6, it works.
Spotted on the list on dependencies that conda uses. Dockerized Jupyter makes just difficult to apply the correct lib without deleting bunch of dirs and restarting the kernel.

What is the dependency of geopandas to this on pip? Because the error was mainly triggered by the following combination:
```
conda install -y -c conda-forge osmnx
pip install geopandas descartes shapely fiona pyproj six
```
This leads to the installation of most recent version of pyproj

**Comment by jorisvandenbossche** [comment]
@aijanai Thanks for opening the issue!
> SOLVED It was pyproj 2.1.3. By setting it to 1.9.6, it works.

OK, but that still means we need to figure out why it was failing with pyproj 2.1.3 (as geopandas is supposed to work with the latest pyproj).

It's a strange one, as that CRS definition that is failing should be totally valid. And I can't reproduce it locally. 
But so originally you installed it with pip? Could you try to install it with conda?

**Comment by jorisvandenbossche** [comment]
Actually, could you show the output of `conda list`? As doing the `conda install -y -c conda-forge osmnx` should already install all of those other packages that you are afterwards installing with pip, so not sure it was actually installed with pip. 
Also, when I do `conda install -y -c conda-forge osmnx` in a clean env, I also get pyproj 1.9.6 and not 2.1.3

**Comment by aijanai** [comment]
Doing it in a clean environment yelds to a working installation:
```
MAGI-SS:~ albertomassidda$ conda create --name test python=3.7 osmnx
WARNING: The conda.compat module is deprecated and will be removed in a future release.
Collecting package metadata: done
Solving environment: done


==> WARNING: A newer version of conda exists. <==
  current version: 4.6.11
  latest version: 4.6.14

Please update conda by running

    $ conda update -n base -c defaults conda



## Package Plan ##

  environment location: /anaconda3/envs/cerved

  added / updated specs:
    - osmnx
    - python=3.7


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    altair-3.0.1               |           py37_0         726 KB  conda-forge
    cffi-1.12.3                |   py37hb5b8e2f_0         214 KB
    cligj-0.5.0                |           py37_0          12 KB
    descartes-1.1.0            |           py37_0           9 KB
    fiona-1.8.4                |   py37h9a122fd_0         932 KB
    gdal-2.3.3                 |   py37hbe65578_0         1.3 MB
    geopandas-0.4.1            |             py_0         891 KB
    jinja2-2.10.1              |           py37_0         188 KB
    kiwisolver-1.1.0           |   py37h0a44026_0          60 KB
    mapclassify-2.0.1          |             py_0          26 KB
    mkl_fft-1.0.12             |   py37h5e564d8_0         155 KB
    munch-2.3.2                |           py37_0          14 KB
    networkx-2.3               |             py_0         1.1 MB
    numpy-1.16.3               |   py37hacdab7b_0          49 KB
    numpy-base-1.16.3          |   py37h6575580_0         4.2 MB
    pip-19.1                   |           py37_0         1.8 MB
    psycopg2-2.7.6.1           |   py37ha12b0ac_0         292 KB
    pyproj-1.9.6               |   py37h9c430a6_0          65 KB
    rtree-0.8.3                |           py37_0          46 KB
    setuptools-41.0.1          |           py37_0         635 KB
    shapely-1.6.4              |   py37he8793f5_0         293 KB
    sqlalchemy-1.3.3           |   py37h1de35cc_0         1.7 MB
    urllib3-1.24.2             |           py37_0         152 KB
    vincent-0.4.4              |             py_1          28 KB  conda-forge
    ------------------------------------------------------------
                                           Total:        14.8 MB

The following NEW packages will be INSTALLED:

  altair             conda-forge/osx-64::altair-3.0.1-py37_0
  asn1crypto         pkgs/main/osx-64::asn1crypto-0.24.0-py37_0
  attrs              pkgs/main/osx-64::attrs-19.1.0-py37_1
  blas               pkgs/main/osx-64::blas-1.0-mkl
  branca             conda-forge/noarch::branca-0.3.1-py_0
  bzip2              pkgs/main/osx-64::bzip2-1.0.6-h1de35cc_5
  ca-certificates    pkgs/main/osx-64::ca-certificates-2019.1.23-0
  cairo              pkgs/main/osx-64::cairo-1.14.12-hc4e6be7_4
  certifi            pkgs/main/osx-64::certifi-2019.3.9-py37_0
  cffi               pkgs/main/osx-64::cffi-1.12.3-py37hb5b8e2f_0
  chardet            pkgs/main/osx-64::chardet-3.0.4-py37_1
  click              pkgs/main/osx-64::click-7.0-py37_0
  click-plugins      pkgs/main/noarch::click-plugins-1.1.1-py_0
  cligj              pkgs/main/osx-64::cligj-0.5.0-py37_0
  cryptography       pkgs/main/osx-64::cryptography-2.6.1-py37ha12b0ac_0
  curl               pkgs/main/osx-64::curl-7.64.1-ha441bb4_0
  cycler             pkgs/main/osx-64::cycler-0.10.0-py37_0
  decorator          pkgs/main/osx-64::decorator-4.4.0-py37_1
  descartes          pkgs/main/osx-64::descartes-1.1.0-py37_0
  entrypoints        pkgs/main/osx-64::entrypoints-0.3-py37_0
  expat              pkgs/main/osx-64::expat-2.2.6-h0a44026_0
  fiona              pkgs/main/osx-64::fiona-1.8.4-py37h9a122fd_0
  folium             conda-forge/noarch::folium-0.8.3-py_0
  fontconfig         pkgs/main/osx-64::fontconfig-2.13.0-h5d5b041_1
  freetype           pkgs/main/osx-64::freetype-2.9.1-hb4e5f40_0
  freexl             pkgs/main/osx-64::freexl-1.0.5-h1de35cc_0
  gdal               pkgs/main/osx-64::gdal-2.3.3-py37hbe65578_0
  geographiclib      conda-forge/noarch::geographiclib-1.49-py_0
  geopandas          pkgs/main/noarch::geopandas-0.4.1-py_0
  geopy              conda-forge/noarch::geopy-1.19.0-py_0
  geos               pkgs/main/osx-64::geos-3.7.1-h0a44026_0
  gettext            pkgs/main/osx-64::gettext-0.19.8.1-h15daf44_3
  giflib             pkgs/main/osx-64::giflib-5.1.4-h1de35cc_1
  glib               pkgs/main/osx-64::glib-2.56.2-hd9629dc_0
  hdf4               pkgs/main/osx-64::hdf4-4.2.13-h39711bb_2
  hdf5               pkgs/main/osx-64::hdf5-1.10.4-hfa1e0ec_0
  icu                pkgs/main/osx-64::icu-58.2-h4b95b61_1
  idna               pkgs/main/osx-64::idna-2.8-py37_0
  intel-openmp       pkgs/main/osx-64::intel-openmp-2019.3-199
  jinja2             pkgs/main/osx-64::jinja2-2.10.1-py37_0
  jpeg               pkgs/main/osx-64::jpeg-9b-he5867d9_2
  json-c             pkgs/main/osx-64::json-c-0.13.1-h3efe00b_0
  jsonschema         pkgs/main/osx-64::jsonschema-3.0.1-py37_0
  kealib             pkgs/main/osx-64::kealib-1.4.7-hf5ed860_6
  kiwisolver         pkgs/main/osx-64::kiwisolver-1.1.0-py37h0a44026_0
  krb5               pkgs/main/osx-64::krb5-1.16.1-hddcf347_7
  libboost           pkgs/main/osx-64::libboost-1.67.0-hebc422b_4
  libcurl            pkgs/main/osx-64::libcurl-7.64.1-h051b688_0
  libcxx             pkgs/main/osx-64::libcxx-4.0.1-hcfea43d_1
  libcxxabi          pkgs/main/osx-64::libcxxabi-4.0.1-hcfea43d_1
  libdap4            pkgs/main/osx-64::libdap4-3.19.1-h3d3e54a_0
  libedit            pkgs/main/osx-64::libedit-3.1.20181209-hb402a30_0
  libffi             pkgs/main/osx-64::libffi-3.2.1-h475c297_4
  libgdal            pkgs/main/osx-64::libgdal-2.3.3-h0950a36_0
  libgfortran        pkgs/main/osx-64::libgfortran-3.0.1-h93005f0_2
  libiconv           pkgs/main/osx-64::libiconv-1.15-hdd342a3_7
  libkml             pkgs/main/osx-64::libkml-1.3.0-hbe12b63_4
  libnetcdf          pkgs/main/osx-64::libnetcdf-4.6.1-hd5207e6_2
  libpng             pkgs/main/osx-64::libpng-1.6.37-ha441bb4_0
  libpq              pkgs/main/osx-64::libpq-11.2-h051b688_0
  libspatialindex    pkgs/main/osx-64::libspatialindex-1.8.5-h2c08c6b_2
  libspatialite      pkgs/main/osx-64::libspatialite-4.3.0a-h644ec7d_19
  libssh2            pkgs/main/osx-64::libssh2-1.8.2-ha12b0ac_0
  libtiff            pkgs/main/osx-64::libtiff-4.0.10-hcb84e12_2
  libxml2            pkgs/main/osx-64::libxml2-2.9.9-hab757c2_0
  mapclassify        pkgs/main/noarch::mapclassify-2.0.1-py_0
  markupsafe         pkgs/main/osx-64::markupsafe-1.1.1-py37h1de35cc_0
  matplotlib         pkgs/main/osx-64::matplotlib-3.0.3-py37h54f8f79_0
  mkl                pkgs/main/osx-64::mkl-2019.3-199
  mkl_fft            pkgs/main/osx-64::mkl_fft-1.0.12-py37h5e564d8_0
  mkl_random         pkgs/main/osx-64::mkl_random-1.0.2-py37h27c97d8_0
  munch              pkgs/main/osx-64::munch-2.3.2-py37_0
  ncurses            pkgs/main/osx-64::ncurses-6.1-h0a44026_1
  networkx           pkgs/main/noarch::networkx-2.3-py_0
  numpy              pkgs/main/osx-64::numpy-1.16.3-py37hacdab7b_0
  numpy-base         pkgs/main/osx-64::numpy-base-1.16.3-py37h6575580_0
  openjpeg           pkgs/main/osx-64::openjpeg-2.3.0-hb95cd4c_1
  openssl            pkgs/main/osx-64::openssl-1.1.1b-h1de35cc_1
  osmnx              conda-forge/noarch::osmnx-0.9-py_1
  pandas             pkgs/main/osx-64::pandas-0.24.2-py37h0a44026_0
  pcre               pkgs/main/osx-64::pcre-8.43-h0a44026_0
  pip                pkgs/main/osx-64::pip-19.1-py37_0
  pixman             pkgs/main/osx-64::pixman-0.38.0-h1de35cc_0
  poppler            pkgs/main/osx-64::poppler-0.65.0-ha097c24_1
  poppler-data       pkgs/main/osx-64::poppler-data-0.4.9-0
  proj4              pkgs/main/osx-64::proj4-5.2.0-h0a44026_1
  psycopg2           pkgs/main/osx-64::psycopg2-2.7.6.1-py37ha12b0ac_0
  pycparser          pkgs/main/osx-64::pycparser-2.19-py37_0
  pyopenssl          pkgs/main/osx-64::pyopenssl-19.0.0-py37_0
  pyparsing          pkgs/main/noarch::pyparsing-2.4.0-py_0
  pyproj             pkgs/main/osx-64::pyproj-1.9.6-py37h9c430a6_0
  pyrsistent         pkgs/main/osx-64::pyrsistent-0.14.11-py37h1de35cc_0
  pysocks            pkgs/main/osx-64::pysocks-1.6.8-py37_0
  python             pkgs/main/osx-64::python-3.7.3-h359304d_0
  python-dateutil    pkgs/main/osx-64::python-dateutil-2.8.0-py37_0
  pytz               pkgs/main/noarch::pytz-2019.1-py_0
  readline           pkgs/main/osx-64::readline-7.0-h1de35cc_5
  requests           pkgs/main/osx-64::requests-2.21.0-py37_0
  rtree              pkgs/main/osx-64::rtree-0.8.3-py37_0
  scipy              pkgs/main/osx-64::scipy-1.2.1-py37h1410ff5_0
  setuptools         pkgs/main/osx-64::setuptools-41.0.1-py37_0
  shapely            pkgs/main/osx-64::shapely-1.6.4-py37he8793f5_0
  six                pkgs/main/osx-64::six-1.12.0-py37_0
  sqlalchemy         pkgs/main/osx-64::sqlalchemy-1.3.3-py37h1de35cc_0
  sqlite             pkgs/main/osx-64::sqlite-3.28.0-ha441bb4_0
  tk                 pkgs/main/osx-64::tk-8.6.8-ha441bb4_0
  toolz              pkgs/main/osx-64::toolz-0.9.0-py37_0
  tornado            pkgs/main/osx-64::tornado-6.0.2-py37h1de35cc_0
  urllib3            pkgs/main/osx-64::urllib3-1.24.2-py37_0
  vincent            conda-forge/noarch::vincent-0.4.4-py_1
  wheel              pkgs/main/osx-64::wheel-0.33.1-py37_0
  xerces-c           pkgs/main/osx-64::xerces-c-3.2.2-h44e365a_0
  xz                 pkgs/main/osx-64::xz-5.2.4-h1de35cc_4
  zlib               pkgs/main/osx-64::zlib-1.2.11-h1de35cc_3
  zstd               pkgs/main/osx-64::zstd-1.3.7-h5bba6e5_0
```

Note that geopandas is not the latest, nor pyproj:
```
  geopandas          pkgs/main/noarch::geopandas-0.4.1-py_0
  pyproj             pkgs/main/osx-64::pyproj-1.9.6-py37h9c430a6_0
```

But when I started my project, I just used `pip install geopandas descartes shapely fiona pyproj six`. This yelded almost certainly to download geopandas==0.5 and pyproj=2.1.3. When I added `conda`on top of it, it did not redownload the correct lib version since it already had a more recent one.

**Comment by aijanai** [comment]
Anyway, yes, it's a mistery on why pyproj failed since now I can't reproduce it:

Steps:
1) Installs on a clean env all latest and greatest:
```
conda create --name pyproj_fail python=3
conda activate pyproj_fail
pip install geopandas descartes shapely fiona pyproj six
```

Execute following:
```
#!/usr/bin/env python
# coding: utf-8

from shapely.geometry import Point
import geopandas

s = geopandas.GeoSeries([Point(12.47312, 41.83586)])
s.crs = {'init': 'epsg:4326', 'no_defs': True}
print(s)

s=s.to_crs(epsg=31370)

print(s)
```
It used to fail, now it runs fine :|

Can't reproduce the steps I followed to mess that up to the point of crashing that way

**Comment by jorisvandenbossche** [comment]
@aijanai Thanks for trying to reproduce it again!

**Comment by aijanai** [comment]
My duty.

You library rocks the ages!

**Comment by Herjar** [comment]
I had the same issue and also fixed it by installing pyproj 1.9.6, but I am not using conda.

**Comment by jorisvandenbossche** [comment]
@Herjar can you still reproduce the error? 
(as the error should not happen. If you can find a way to reproduce it, that would be very welcome. Because if it actually happens in some circumstances, we should try to fix it)

**Comment by Sorgann** [comment]
I created a docker with this Dockerfile
```
FROM jupyter/tensorflow-notebook

# Install geo python packages
RUN conda install --quiet --yes \
            'gdal=2.4*' \
            'geopandas=0.3*' && \
            conda clean -tipsy && \
            fix-permissions $CONDA_DIR && \
            fix-permissions /home/$NB_USER

RUN pip install dvc

RUN  conda install -c conda-forge tifffile
```
Pyproj version is `Version: 2.1.3`

If I use this jupyter and I try :
```
#!/usr/bin/env python
# coding: utf-8

from shapely.geometry import Point
import geopandas

s = geopandas.GeoSeries([Point(12.47312, 41.83586)])
s.crs = {'init': 'epsg:4326', 'no_defs': True}

s=s.to_crs(epsg=31370)
```

I have this error :
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/opt/conda/lib/python3.7/site-packages/geopandas/geoseries.py", line 285, in to_crs
    proj_in = pyproj.Proj(self.crs, preserve_units=True)
  File "/opt/conda/lib/python3.7/site-packages/pyproj/proj.py", line 295, in __init__
    self.crs = CRS.from_user_input(projparams if projparams is not None else kwargs)
  File "/opt/conda/lib/python3.7/site-packages/pyproj/crs.py", line 224, in from_user_input
    return cls(**value)
  File "/opt/conda/lib/python3.7/site-packages/pyproj/crs.py", line 146, in __init__
    super(CRS, self).__init__(projstring)
  File "pyproj/_crs.pyx", line 317, in pyproj._crs._CRS.__init__
pyproj.exceptions.CRSError: Invalid projection: +init=epsg:4326 +no_defs +type=crs: (Internal Proj Error: proj_create: cannot expand +init=epsg:4326 +no_defs +type=crs)
```

I hope it can help you to reproduce this error.

**Comment by snowman2** [comment]
With pyproj>=2.0 you need to remove the `no_defs` when using `init=epsg:..` syntax. I don't know exactly why, just that it is the case.

**Comment by piyathilaka8560** [comment]
Today is 2023.02.02. This issue still already have. 
I am also facing this issue.

please tell me if you have any solution to solve this with the following library versions.
sys.version : 3.9.13 (main, Aug 25 2022, 23:51:50) [MSC v.1916 64 bit (AMD64)]
pyproj.__version__: 2.6.1.post1
geopandas.__version__: 0.9.0

My error is as follows,
CRSError: Invalid projection: epsg:4326: (Internal Proj Error: proj_create: SQLite error on SELECT name, type, coordinate_system_auth_name, coordinate_system_code, datum_auth_name, datum_code, area_of_use_auth_name, area_of_use_code, text_definition, deprecated FROM geodetic_crs WHERE auth_name = ? AND code = ?: no such column: area_of_use_auth_name)

**Comment by snowman2** [comment]
@piyathilaka8560, this is the error you are running into: https://pyproj4.github.io/pyproj/stable/gotchas.html#internal-proj-error-sqlite-error-on-select . It means your environment has multiple versions of PROJ and the active version cannot find the correct database.

**Comment by piyathilaka8560** [comment]
@snowman2, Thanks for your comment.

As the solution, the following method worked for me.

I found that pyproj version has this bug and I tried as follows as a solution for that.

Without Anaconda, I tested with Pycharm to install the latest geopandas with pyproj
pip install geopandas (in pycharm terminal)

sys.version : 3.10.4 (tags/v3.10.4:[9d38120](https://bitbucket.org/piyathilaka/geopanda_with_gdal/commits/9d38120), Mar 23 2022, 23:13:41) [MSC v.1929 64 bit (AMD64)]
pyproj.version : 3.4.1
geopandas.version: 0.12.2
